lite/dbprovider.go

```diff
@@ -258,14 +258,14 @@ func parseKey(key []byte) (chainID string, height int64, part string, ok bool) {
 }
 
 func parseSignedHeaderKey(key []byte) (chainID string, height int64, ok bool) {
-	chainID, height, part, ok := parseKey(key)
+	chainID, height, part, _ := parseKey(key)
 	if part != "sh" {
 		return "", 0, false
 	}
 	return chainID, height, true
 }
 
 func parseChainKeyPrefix(key []byte) (chainID string, height int64, ok bool) {
-	chainID, height, _, ok = parseKey(key)
+	chainID, height, _, _ = parseKey(key)
```

Comment:
I'm surprised that we do not check `ok` here though. If `keyPattern.FindSubmatch(key)` doesn't match anything this will return `"", 0, "", false`.

Reply:
Do you think we should return `ok` instead of `true`?

Reply:
https://github.com/tendermint/tendermint/pull/3386/commits/77379e0cda71616da5cf32b25d44c1eceb80c869